internal/updater/updater_test.go

```diff
@@ -0,0 +1,457 @@
+					Reason:             "started",
+					Message:            "starting up",
+				},
+			},
+		},
+	}
+)
+
+func NewFakeUpdater(t *testing.T) Updater {
+	scheme := runtime.NewScheme()
+	err := schemeBuilder.AddToScheme(scheme)
+	require.Equal(t, nil, err)
+
+	return Updater{
+		client: fake.NewClientBuilder().WithObjects(bundle).WithScheme(scheme).Build(),
+	}
+}
+
+func TestApply(t *testing.T) {
+	for _, tt := range []struct {
+		description string
+		arg         *rukpakv1alpha1.BundleStatus
+	}{
+		{
+			description: "No changes",
+			arg: &rukpakv1alpha1.BundleStatus{
+				Info:               &rukpakv1alpha1.BundleInfo{},
+				Phase:              rukpakv1alpha1.PhaseFailing,
+				Digest:             "digest",
+				ObservedGeneration: 1,
+				Conditions: []metav1.Condition{
+					{
+						Type:               "Working",
+						Status:             metav1.ConditionTrue,
+						ObservedGeneration: 3,
+						LastTransitionTime: metav1.Time{},
+						Reason:             "requested",
+						Message:            "Working correctly",
+					},
+					{
+						Type:               "starting",
+						Status:             metav1.ConditionTrue,
+						ObservedGeneration: 1,
+						LastTransitionTime: metav1.Time{},
+						Reason:             "started",
+						Message:            "starting up",
+					},
+				},
+			},
+		},
+		{
+			description: "Happy path",
+			arg: &rukpakv1alpha1.BundleStatus{
+				Info:               &rukpakv1alpha1.BundleInfo{},
+				Phase:              rukpakv1alpha1.PhaseFailing,
+				Digest:             "digest1",
+				ObservedGeneration: 3,
+				Conditions: []metav1.Condition{
+					{
+						Type:               "Complated",
+						Status:             metav1.ConditionTrue,
+						ObservedGeneration: 3,
+						LastTransitionTime: metav1.Time{},
+						Reason:             "requested",
+						Message:            "finished correctly",
+					},
+				},
+			},
+		},
+	} {
+		t.Run(tt.description, func(t *testing.T) {
+			u := NewFakeUpdater(t)
+			u.UpdateStatus(EnsureCondition(tt.arg.Conditions[0]), EnsureObservedGeneration(tt.arg.ObservedGeneration), EnsureBundleDigest(tt.arg.Digest), SetBundleInfo(tt.arg.Info), SetPhase(tt.arg.Phase))
+
+			argbundle := &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "olm-crds",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: "core.rukpak.io/plain",
+					Image:                "quay.io/tflannag/olm-plain-bundle:olm-crds-v0.20.0",
+				},
+				Status: *tt.arg,
+			}
+			result := u.Apply(context.Background(), argbundle)
+			require.Equal(t, nil, result)
+			b := rukpakv1alpha1.Bundle{}
+			err := u.client.Get(context.Background(), client.ObjectKeyFromObject(argbundle), &b)
+			require.Equal(t, nil, err)
+			require.Equal(t, tt.arg.Info, b.Status.Info)
+			require.Equal(t, tt.arg.Phase, b.Status.Phase)
+			require.Equal(t, tt.arg.Digest, b.Status.Digest)
+			require.Equal(t, tt.arg.ObservedGeneration, b.Status.ObservedGeneration)
```

Comment:
nit: Does this work here? It would cut down a few lines for us. Referencing[ this post ](https://stackoverflow.com/questions/18208394/how-to-test-the-equivalence-of-maps-in-golang)talking about how comparison between two maps (the Status here) can be guaranteed this way.

```suggestion
			require.Equal(fmt.Sprint(tt.arg), fmt.Sprint(b.Status)
```

Reply:
Thanks for the suggestion.  I tried but it didn't work here.  One issue is the `Info` is a pointer so the Sprintf prints the address instead of the contents.  The other issue is the `Conditions`.  It's an array.  It may have additional elements and the order of the elements should not be the matter.